### PR TITLE
Fix GC polls around Thread.Spinwait

### PR DESCRIPTION
**Pull request by jkotas**
Avoid GC starvation when Thread.Spinwait is called in a tight loop

### CODE REVIEW

File: src/coreclr/nativeaot/System.Private.CoreLib/src/Internal/Runtime/Augments/RuntimeAugments.cs [hunk 1]
```diff
@@ -1061,11 +1061,6 @@ public static void RhHandleFree(IntPtr handle)
             RuntimeImports.RhHandleFree(handle);
         }
 
-        public static IntPtr RhGetOSModuleForMrt()
```
Comment (jkotas) [inline src/coreclr/nativeaot/System.Private.CoreLib/src/Internal/Runtime/Augments/RuntimeAugments.cs:1064]: Unrelated cleanup

**Review by MichalStrehovsky — APPROVED** [review]
Thank you!
